lib/streamlit/commands/execution_control.py

```diff
@@ -87,3 +90,49 @@ def experimental_rerun() -> NoReturn:
     # be seen.
     _LOGGER.warning(msg)
     rerun()
+
+
+@gather_metrics("switch_page")
+def switch_page(page: str) -> NoReturn:
+    """Switch the current programmatically page in a multi-page app.
+    When `st.switch_page()` is called with a page, the current page script is halted
+    and the requested page script will be queued to run from the top.
+    Parameters
+    ----------
+    page: str
+        The file path, relative to the main script, of the page to switch to.
+    """
+
+    ctx = get_script_run_ctx()
+
+    query_string = ""
+    ctx_main_script = ""
+    if ctx:
+        query_string = ctx.query_string
+        ctx_main_script = ctx.main_script_path
+
+    main_script_path = os.path.join(os.getcwd(), ctx_main_script)
+    main_script_directory = os.path.dirname(main_script_path)
+
+    if page.startswith("./"):
+        page = page[2:]
+    elif page.startswith("/"):
+        page = page[1:]
+
+    requested_page = os.path.join(main_script_directory, page)
```

Comment:
Would appreciate any feedback on this path handling logic - feel like it can probably be improved 😓